dev-util/android-studio/android-studio-3.3.1.0.182.5264788.ebuild

```diff
@@ -0,0 +1,115 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+inherit eutils java-pkg-2 eapi7-ver
+
+RESTRICT="strip"
+QA_PREBUILT="opt/${PN}/bin/libbreakgen*.so
+	opt/${PN}/bin/fsnotifier*
+	opt/${PN}/bin/lldb/*
+	opt/${PN}/lib/libpty/linux/x86*/libpty.so
+	opt/${PN}/plugins/android/lib/libwebp_jni*.so
+	opt/${PN}/plugins/android/resources/perfa/*/libperfa.so
+	opt/${PN}/plugins/android/resources/perfd/*/perfd
+	opt/${PN}/plugins/android/resources/simpleperf/*/simpleperf"
+
+VER_CMP=( $(ver_rs 1- ' ') )
+if [[ ${#VER_CMP[@]} -eq 6 ]]; then
+	STUDIO_V=$(ver_cut 1-4)
+	BUILD_V=$(ver_cut 5-6)
+else
+	STUDIO_V=$(ver_cut 1-3)
+	BUILD_V=$(ver_cut 4-5)
+fi
+
+DESCRIPTION="A new Android development environment based on IntelliJ IDEA"
+HOMEPAGE="http://developer.android.com/sdk/installing/studio.html"
+SRC_URI="https://dl.google.com/dl/android/studio/ide-zips/${STUDIO_V}/${PN}-ide-${BUILD_V}-linux.zip"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+IUSE="selinux"
+KEYWORDS="~amd64 ~x86"
+
+DEPEND="app-arch/zip
+	dev-java/commons-logging:0
+	dev-java/log4j:0"
+
+#	dev-java/guava:18
+RDEPEND=">=virtual/jdk-1.7
+	selinux? ( sec-policy/selinux-android )
+	>=app-arch/bzip2-1.0.6-r4
+	dev-java/commons-logging:0
+	dev-java/log4j:0
+	>=dev-libs/expat-2.1.0-r3
+	>=dev-libs/libffi-3.0.13-r1
+	>=media-libs/fontconfig-2.10.92
+	>=media-libs/freetype-2.5.5
+	>=media-libs/libpng-1.2.51
+	>=media-libs/mesa-10.2.8
+	|| ( gnome-extra/zenity kde-apps/kdialog x11-apps/xmessage x11-libs/libnotify )
+	>=sys-libs/ncurses-5.9-r3:5/5[tinfo]
+	>=sys-libs/zlib-1.2.8-r1
+	>=x11-libs/libX11-1.6.2
+	>=x11-libs/libXau-1.0.7-r1
+	>=x11-libs/libXdamage-1.1.4-r1
+	>=x11-libs/libXdmcp-1.1.1-r1
+	>=x11-libs/libXext-1.3.2
+	>=x11-libs/libXfixes-5.0.1
+	>=x11-libs/libXrender-0.9.8
+	>=x11-libs/libXxf86vm-1.1.3
+	>=x11-libs/libdrm-2.4.46
+	>=x11-libs/libxcb-1.9.1
+	>=x11-libs/libxshmfence-1.1"
+S=${WORKDIR}/${PN}
+PATCHES=( "${FILESDIR}/0001-use-java-home-before-bundled.patch" )
+
+src_prepare() {
+	eapply "${PATCHES[@]}"
```

Comment:
You can just call `default` in here. It applies everything in `PATCHES` array, and also calls `eapply_user`. So you can then remove the eapply_user part below.